### PR TITLE
Docs: implemented typedoc + vitepress documentation

### PR DESCRIPTION
**Pull request by JustSamuel**
# Description
This PR does not add any documentation  by itself, apart from a single page on how the documentation works. The main goal of this PR is to create the base template / structure for creating a documentation site which is automatically deployed. 

[See for yourself here.](https://sudosos.test.gewis.nl/docs/)

## Related issues/external references
#285 

## Types of changes
- Documentation improvement

### CODE REVIEW

File: src/controller/debtor-controller.ts [hunk 1]
```diff
@@ -18,6 +18,18 @@
...
+ * This is the module page of debtor-controller.
+ *
+ * @module debtors
+ */
+
+/**
+ * This is the module page of debtor-controller.
+ *
+ * @module debtors
+ */
```
Comment (Yoronex) [inline src/controller/debtor-controller.ts:31]: Two definitions

File: src/controller/event-shift-controller.ts [hunk 2]
```diff
@@ -18,6 +18,12 @@
  *  @license
  */
 
+/**
+ * This is the module page of event-controller.
```
Comment (Yoronex) [inline src/controller/event-shift-controller.ts:22]: This is the `event-shift-controller`?

File: src/controller/response/debtor-response.ts [hunk 3]
```diff
@@ -18,6 +18,24 @@
...
+ * This is the module page of debtor-response.
+ *
+ * @module debtors
+ */
+
+/**
+ * This is the module page of debtor-response.
+ *
+ * @module debtors
+ */
```
Comment (Yoronex) [inline src/controller/response/debtor-response.ts:37]: Wow three times? Damn

File: src/controller/response/dinero-response.ts [hunk 4]
```diff
@@ -18,6 +18,18 @@
...
+ * This is the module page of the dinero-response.
+ *
+ * @module helpers/dinero
+ */
+
+/**
+ * This is the module page of the dinero-response.
+ *
+ * @module helpers/dinero
+ */
```
Comment (Yoronex) [inline src/controller/response/dinero-response.ts:31]: Twice

File: src/controller/response/event-response.ts [hunk 5]
```diff
@@ -18,6 +18,18 @@
...
+ * This is the module page of the event-response.
+ *
+ * @module events
+ */
+
+/**
+ * This is the module page of the event-response.
+ *
+ * @module events
+ */
```
Comment (Yoronex) [inline src/controller/response/event-response.ts:31]: Twice

File: src/controller/response/financial-mutation-response.ts [hunk 6]
```diff
@@ -18,6 +18,17 @@
...
  */
 
+/**
+ * @hidden
+ * @module
+ */
+
+/**
+ * This is the module page of the financial-mutation-response.
+ * TODO change
```
Comment (Yoronex) [inline src/controller/response/financial-mutation-response.ts:28]: TODO

File: src/controller/response/update-key-response.ts [hunk 7]
```diff
@@ -18,6 +18,12 @@
  *  @license
  */
 
+/**
+ * This is the module page of the update-key-response.
+ *
+ * @module internal/controllers
```
Comment (Yoronex) [inline src/controller/response/update-key-response.ts:24]: If you would have checked the contents of this file, you would have seen this response is used only in the `user-controller` when requesting a new API key. So either `users` or `authentication`.

File: src/controller/user-controller.ts [hunk 8]
```diff
@@ -18,6 +18,18 @@
...
+ * This is the module page of user-controller.
+ *
+ * @module users
+ */
+
+/**
+ * This is the module page of user-controller.
+ *
+ * @module users
+ */
```
Comment (Yoronex) [inline src/controller/user-controller.ts:31]: Twice

File: src/cron.ts [hunk 9]
```diff
@@ -18,6 +18,12 @@
  *  @license
  */
 
+/**
+ * This is the module page the cron job.
```
Comment (Yoronex) [inline src/cron.ts:22]: Nice inglish

File: package.json [hunk 10]
```diff
@@ -51,9 +56,11 @@
     "stripe": "^16.1.0",
     "swagger-model-validator": "^3.0.21",
     "swagger-ui-express": "^5.0.0",
+    "typedoc": "^0.26.7",
```
Comment (Yoronex) [inline package.json:59]: It seems we only use `typedoc` in development environments and never in production environments, so maybe move it to the dev dependencies?

**Review by Yoronex — CHANGES_REQUESTED** [review]
I absolutely hated reviewing this PR... I spent more than 20 minutes on it, but at least 80% of all my comments were simple typos or errors you could have easily found yourself. I even skipped commenting on every documentation page text that could do with improvement, like in `src/database/schema.ts`. So for the next time, please take some time to review your own PRs, so you can save the other some time...

**Comment by JustSamuel** [comment]
> I absolutely hated reviewing this PR... I spent more than 20 minutes on it, but at least 80% of all my comments were simple typos or errors you could have easily found yourself. I even skipped commenting on every documentation page text that could do with improvement, like in src/database/schema.ts. So for the next time, please take some time to review your own PRs, so you can save the other some time...

I really tried to align all the modules, but given that it were around 400 pages it does not sruprise me that it didn't went as smooth. I mainly focused on the non-internal tags I have to admit.

File: package.json [hunk 11]
```diff
@@ -22,7 +22,12 @@
     "lint": "eslint package.json src test --ext .js --ext .ts",
     "lint-fix": "npm run lint -- --fix",
     "typeorm": "node --require ts-node/register ./node_modules/typeorm/cli.js",
-    "cli": "ts-node ./src/gewis/service/cli-service.ts"
+    "cli": "ts-node ./src/gewis/service/cli-service.ts",
+    "predocs:dev": "typedoc",
+    "typedoc": "typedoc",
```
[the remark this answers is not in the document]
Comment (JustSamuel) [reply package.json:27]: pre<x> is called before <x>, this allows for just typedoc gen

File: src/gewis/gewis.ts [hunk 12]
```diff
@@ -18,6 +18,13 @@
  *  @license
  */
 
+/**
+ * This is the module page of GEWIS.
+ *
+ * @module GEWIS
+ * @mergeTarget
```
[the remark this answers is not in the document]
Comment (JustSamuel) [reply src/gewis/gewis.ts:25]: Set which module text will be displayed, in general, all other module text will be ignored as there is only one module. 

The module is the "category"  and the one with @mergeTarget is the category description page.

File: src/controller/event-shift-controller.ts (hunk 2 above)
Comment (Yoronex) [reply src/controller/event-shift-controller.ts:22]: Bruh

**Review by Yoronex — CHANGES_REQUESTED** [review]

**Review by Yoronex — APPROVED** [review]

**Pull request merged**